plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -399,6 +402,7 @@ private Optional<DataFileInfo> rewriteParquetFile(Path path, FileDeletion deleti
             throws IOException
     {
         LongBitmapDataProvider rowsToDelete = deletion.rowsToDelete;
+        LongBitmapDataProvider rowsDeletedBecauseOfUpdate = deletion.rowsDeletedBecauseOfUpdate();
```

Comment:
Can we avoid marking update preimage rows in both Bitmaps?

Reply:
i don't think so, we need to carry information if particular position deletion came from DELETE or from UPDATE, instead of a bitmap a normal Map could be used that would store the reason for the deletion but I assumed it is more efficient to use 2 bitmaps.
I can change to one Map if you think it is better

Reply:
Two Bitmaps is good, but you have redundant information for updated rows, the positions for both `rowsToDelete` and `rowsDeletedBecauseOfUpdate` are set. I'd expect you to set one or the other, not both.

Reply:
ok I think I figured out your point, and now the second if would probably also work the way you wrote it, I will give it a shot. thanks